app/models/solid_queue/record.rb

```diff
@@ -6,6 +6,11 @@ class Record < ActiveRecord::Base
 
     connects_to(**SolidQueue.connects_to) if SolidQueue.connects_to
 
+    def self.clear_all_connections!
+      self.connection_handler.clear_all_connections!(:writing)
+      self.connection_handler.clear_all_connections!(:reading)
```

Comment:
I'm curious why two calls are being made for reading and writing roles here.

By default, if no arguments are passed to `ConnectionHandler#clear_all_connections!` it should clear all roles. Did you see something unexpected that led you to code it this way?

Reply:
Hello, to be honest I was just copy pasting from another app without digging the reason.
The behavior for this method changes in rails 7.2 according to [this PR](https://github.com/rails/rails/commit/64cbcd7a8c0d83f0da48fdd01bd156aaf12c8221#diff-d1e52309089a1c9135a7d1bb82bdfdf7410d7d7a294013e16e71f0171154eefbL196-L199). I just checked an solid queue still supports rails 7.1, so that would ensure compatibility.

Reply:
@alxckn Ah! That's a very good reason to do it that way. Thanks for the explanation! 👍